packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -380,8 +404,8 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
 
         if (hasEIP1559Support(this.chain)) {
             return {
-                maxFeePerGas: maxFeePerGas ? BigInt(maxFeePerGas.toString()) : undefined,
-                maxPriorityFeePerGas: maxPriorityFeePerGas ? BigInt(maxPriorityFeePerGas.toString()) : undefined,
+                ...(maxFeePerGas && { maxFeePerGas: BigInt(maxFeePerGas.toString()) }),
+                ...(maxPriorityFeePerGas && { maxPriorityFeePerGas: BigInt(maxPriorityFeePerGas.toString()) }),
```

Comment:
this caught me off guard because I'm thinking we should do `maxFeePerGas != null && ...` but that's not true, we do want to ignore if it's 0 also
so maybe drop a comment
```suggestion
                // only include if non-null and non-zero
                ...(maxFeePerGas && { maxFeePerGas: BigInt(maxFeePerGas.toString()) }),
                ...(maxPriorityFeePerGas && { maxPriorityFeePerGas: BigInt(maxPriorityFeePerGas.toString()) }),
```